### PR TITLE
term-entry: kotlin filter()

### LINKED ISSUE #3725: [Term Entry] Kotlin Strings .filter()
### Reason/inspiration (optional)

We would like a new term entry on the `filter()` method for Kotlin: [Strings](https://www.codecademy.com/resources/docs/kotlin/strings). The entry should go in a new file under `docs/content/kotlin/concepts/strings/terms/filter/filter.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by vrun1208**
### Description
closes #3725 
added a new term entry for Kotlin strings .filter() method.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+```pseudo
+str.filter(predicate: (char) -> Boolean)
+
+```
+
+Given a string `str` above, `filter()` method can filter the characters from the string using a predicate which is passed as an argument.
+
+### Predicate
+
+Predicates are lambda functions that takes an element of the collection as an argument and returns a `Boolean` value. The filter() function will include the elements for which the predicate returns `true` in the resulting collection.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:29]:
```suggestion

- `Predicates` are lambda functions that takes an element of the collection as an argument and returns a `Boolean` value. The **`filter()`** function will include the elements for which the predicate returns `true` in the resulting collection.
```
No need to create a heading for this.

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Syntax
+
+The **filter()** function is used to filter elements from a collection based on a given _predicate_. It returns a new collection containing only elements which statifies a specified condition.
+
+```pseudo
+str.filter(predicate: (char) -> Boolean)
+
+```
+
+Given a string `str` above, `filter()` method can filter the characters from the string using a predicate which is passed as an argument.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:25]:
```suggestion
- When working with the string `str` mentioned above, the `filter()` method allows selective removal of characters from the string based on a provided predicate.
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+Title: 'filter()'
+Description: 'filter elements from collection based on a given predicate.'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:3]:
```suggestion
Description: 'Filters elements from collection based on a given predicate.'
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:6]:
```suggestion
  - 'Code Foundations'
```
Web dev doesn't make sense here.

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Filter'
+  - 'Strings'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+## Syntax
+
+The **filter()** function is used to filter elements from a collection based on a given _predicate_. It returns a new collection containing only elements which statifies a specified condition.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:18]:
```suggestion
The **`filter()`** function is used to filter elements from a collection based on a given `predicate`. It returns a new collection containing only elements which satisfies a specified condition.
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+```pseudo
+str.filter(predicate: (char) -> Boolean)
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:22]:
```suggestion
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+```kotlin
+fun main(args: Array<String>) {
+    val str = "Hello, World!"
+    val result = str.filter({ it -> it.isLetter() })
+    println("Filtered String : " + result)
+}
+```
+
+Output of the above code will be -
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:43]:
```suggestion
Output of the above code will be:
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+```pseudo
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:45]:
```suggestion
```shell
```
use the tag shell for code output

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
+---
+Title: 'filter()'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:2]:
```suggestion
Title: '.filter()'
```

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Collections'
+  - 'Filter'
+  - 'Strings'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+## Syntax
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:17]:
```suggestion

```
this should not come here.

File: content/kotlin/concepts/strings/terms/filter/filter.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+The **filter()** function is used to filter elements from a collection based on a given _predicate_. It returns a new collection containing only elements which statifies a specified condition.
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/filter/filter.md:19]:
```suggestion
## Syntax 

```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Hey 👋 for not commit these changes and after that I will give further suggestions.

File: content/kotlin/concepts/strings/terms/filter/filter.md (hunk 2 above)
Comment (vrun1208) [reply content/kotlin/concepts/strings/terms/filter/filter.md:25]: @dakshdeepHERE, `filter()` method does not remove the elements from the string instead it creates a new collection which includes only selective elements

Comment (dakshdeepHERE) [reply content/kotlin/concepts/strings/terms/filter/filter.md]: Yes let me refactor it. It basically create a new collection that includes only the elements that satisfy a specified predicate.

Comment (vrun1208) [reply content/kotlin/concepts/strings/terms/filter/filter.md]: yes, I have made the changes as you suggested also this one. You can check it on my latest commit.

**Comment by dakshdeepHERE** [comment]
@vrun1208 LGTM.

**Review by dakshdeepHERE — APPROVED** [review]

**Review by caupolicandiaz — APPROVED** [review]
@vrun1208 LGTM

**Pull request merged**